src/pages/DelegatoryValidator/StakeDialog.tsx

```diff
@@ -5,47 +5,113 @@ import {
   DialogContent,
   DialogTitle,
   FormHelperText,
+  IconButton,
   InputAdornment,
+  List,
+  ListItem,
+  ListItemText,
   OutlinedInput,
+  Stack,
+  Typography,
 } from "@mui/material";
-import React from "react";
-import ContentBox from "../../components/IndividualPageContent/ContentBox";
-import ContentRow from "../../components/IndividualPageContent/ContentRow";
+import React, {useState} from "react";
+import CloseIcon from "@mui/icons-material/Close";
+import {Types} from "aptos";
+import TimestampValue from "../../components/IndividualPageContent/ContentValue/TimestampValue";
+import {grey} from "../../themes/colors/aptosColorPalette";
 
 type StakeDialogProps = {
   handleDialogClose: () => void;
   isDialogOpen: boolean;
+  accountResource?: Types.MoveResource | undefined;
 };
 
 export default function StakeDialog({
   handleDialogClose,
   isDialogOpen,
+  accountResource,
 }: StakeDialogProps) {
+  const lockedUntilSecs = accountResource
+    ? BigInt((accountResource.data as any).locked_until_secs)
+    : null;
+  const [stakedAmount, setStakedAmount] = useState<string>();
+  const [stakedAmountError, setStakedAmountError] = useState<boolean>(false);
+  const handleStakedAmountChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const amt = event.target.value;
+    setStakedAmountError(isNaN(Number(amt)));
+    setStakedAmount(amt);
+  };
+
   return (
-    <Dialog open={isDialogOpen} onClose={handleDialogClose} fullWidth={true}>
-      <DialogTitle>Stake into the pool</DialogTitle>
-      <DialogContent>
-        <FormHelperText>Enter amount</FormHelperText>
-        <OutlinedInput
-          placeholder="0"
-          endAdornment={<InputAdornment position="end">APT</InputAdornment>}
-        />
-      </DialogContent>
-      <ContentBox>
-        <ContentRow title="Operator commission" value={null} />
-        <ContentRow title="Compound rewards" value={null} />
-        <ContentRow title="Next unlock in" value={null} />
-      </ContentBox>
-      <ContentBox>
-        <ContentRow title="Transaction fee" value={null} />
-      </ContentBox>
-      <DialogActions>
-        <Button onClick={handleDialogClose}>Deposit</Button>
-      </DialogActions>
-      <DialogContent>
-        Be aware that you will be able to see your funds in the pool after 1
-        epoch (~2 hours) due to the delay time
-      </DialogContent>
+    <Dialog open={isDialogOpen} onClose={handleDialogClose}>
+      <IconButton
```

Comment:
it's a BE work so we can do it later but it's worth it to create a `StyledDialog` component with this "X" icon in it and get it reused for whenever we need a dialog/modal. For example, wallet button modal should have the same style.